ui/lib/sync/addon/routes.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import buildRoutes from 'ember-engines/routes';
+
+export default buildRoutes(function () {
+  this.route('secrets', function () {
+    this.route('overview');
+    this.route('destinations', function () {
+      this.route('create', function () {
+        this.route('destination');
```

Comment:
I think we decided to make the url here `create/destination/:type` ?
```suggestion
        this.route('destination', { path: '/:type' });
```